src/IOTA/mam.ts

```diff
@@ -1,58 +1,47 @@
-import * as MAM from '@iota/mam';
+import {
+    channelRoot,
+    createChannel,
+    createMessage,
+    mamAttach,
+    mamFetchAll,
+    MamMode,
+    IMamMessage,
+    IMamFetchedMessage,
+    IMamChannelState
+} from '@iota/mam.js';
+import { composeAPI } from '@iota/core';
 import { asciiToTrytes, trytesToAscii } from '@iota/converter';
 
-
 //Settings for a MAM stream. 
 export class MAMSettings {
-    public mode : MAM_MODE;
-    public sideKey : string;
-    public securityLevel : number;
+    public mode: MamMode;
+    public sideKey: string;
+    public securityLevel: number;
 
-    constructor(mode : MAM_MODE = MAM_MODE.PRIVATE, sideKey ?: string, securityLevel : number = 2) {
+    constructor(mode: MamMode = 'private', sideKey?: string, securityLevel: number = 2) {
```

Comment:
The enum was done on purpose to prevent simple spelling mistakes from crashing the application. I'd prefer to keep the enum around instead of typing 'private', 'public' or 'restricted'.

Reply:
Using the `MamMode` type exported through [mam.js](https://github.com/iotaledger/mam.js/blob/master/typings/models/mamMode.d.ts#L1) should also work the same way i.e., won't allow any typos. If you prefer to keep the enum, I'll revert these changes.

Reply:
@JelleMillenaar Updated in https://github.com/iotaledger/identity.ts/pull/15/commits/c7cba091e920a74c6abeb1b755fc8cd77fa3f10c & https://github.com/iotaledger/identity.ts/pull/15/commits/9f83ec07b1c05a0d2ad28c791235bcf44268ea6f